cpptools/src/fjtools/fjtools.cxx

```diff
@@ -178,157 +176,153 @@ namespace PyJettyFJTools
-							if (move_underflow) {
-								if (bin_2 == 0) {
-									x[2] = thn_rebinned->GetAxis(2)->GetBinCenter(1);
-								}  // bin_2 == 0
-								if (bin_3 == 0) {
-									x[3] = thn_rebinned->GetAxis(3)->GetBinCenter(1);
-									std::string name(thn->GetName());
-									if (name.find("matched") != std::string::npos) {
-										if (bin_2 == 0) { content = 1; }
-										else { content = 0; }
-									}
-								}  // bin_3 == 0
-							} else { continue; }  // move_underflow
-						}  // underflow bins
+                            content *= prior_scale_factor_obs(
+								x[3], content, prior_variation_parameter, prior_option);
+
+                        }  // scale prior
+
+                        // If underflow bin, and if move_underflow flag is activated,
+                        // put the contents of the underflow bin into first bin of thn_rebinned
+                        if (bin_2 == 0 || bin_3 == 0) {
+                            if (move_underflow) {
+                                if (bin_2 == 0) {
+                                    x[2] = thn_rebinned->GetAxis(2)->GetBinCenter(1);
+                                }  // bin_2 == 0
+                                if (bin_3 == 0) {
+                                    x[3] = thn_rebinned->GetAxis(3)->GetBinCenter(1);
+                                    std::string name(thn->GetName());
+                                    if (name.find("matched") != std::string::npos) {
+                                        if (bin_2 == 0) { content = 1; }
+                                        else { content = 0; }
+                                    }
+                                }  // bin_3 == 0
+                            } else { continue; }  // move_underflow
+                        }  // underflow bins
 
                         // THn is filled as (x[0], x[1], x[2], x[3])
                         // corresponding e.g. to (pt_det, pt_true, obs_det, obs_true)
-						thn_rebinned->Fill(x, content);
-                        
+                        thn_rebinned->Fill(x, content);
+
                         // RooUnfoldResponse should be filled as (x[0], x[2], x[1], x[3])
                         // corresponding e.g. to (pt_det, obs_det, pt_true, obs_true)
                         if (do_roounfoldresponse) {
                             roounfold_response->Fill(x[0], x[2], x[1], x[3], content);
                         }
 
-					}  // bin_3 loop
-				}  // bin_2 loop
-			}  // bin_1 loop
-		}  // bin_0 loop
-        
+                    }  // bin_3 loop
+                }  // bin_2 loop
+            }  // bin_1 loop
+        }  // bin_0 loop
+
         std::cout << "writing response..." << std::endl;
         TFile f(response_file_name.c_str(), "UPDATE");
         thn_rebinned->Write();
-        if (do_roounfoldresponse) {
-            roounfold_response->Write();
-        }
+        if (do_roounfoldresponse) { roounfold_response->Write(); }
```

Comment:
for the future I propose we avoid these single-line if statements -- it's inconsistent syntax